### PR TITLE
renaming queries renames it's menu item

### PR DESCRIPTION
**Pull request by ulferts**
https://community.openproject.org/work_packages/16720

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-query-menu-item-renaming.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: app/assets/javascripts/angular/layout/menu-item-factory.js [hunk 1]
```diff
@@ -76,9 +76,25 @@ angular.module('openproject.layout')
         if (linkFn) linkFn(scope, menuItem.children('a'), {});
 
         $compile(menuItem)(scope);
-        $animate.enter(menuItem, container);
+        $animate.enter(menuItem, container, previousMenuItem(title));
       });
+    }
+
+    function previousMenuItem(title) {
```
Comment (tessi) [inline app/assets/javascripts/angular/layout/menu-item-factory.js:97]: Maybe a doc-string explaining this function would help. Its not entirely clear (at least to me) that this function takes a title and returns a menu item which has a title alphabetically preceding the given title-- but only until an element with no `type`-(whatever `type` is)-class appears. :)

Comment (ulferts) [inline app/assets/javascripts/angular/layout/menu-item-factory.js:97]: I like to create mystery. But for you, @tessi, I will create the doc.

**Comment by tessi** [comment]
Looks good to me. I'll merge if travis is happy.

**Comment by hschink** [comment]
Travis is happy, so should @tessi.

**Pull request merged**